test/test_vectors.txt

```diff
@@ -0,0 +1,59 @@
+# This test file is formatted as "<lefthand> <operator> <righthand>", seperated by the space character
+# Implementations should ignore lines starting with "#" and lines that have a length of 0
+
+# Basic numeric ordering (lexical string sort fails these)
+10 > 2
+100 > 10
+
+# Trivial common numerics
+1.0 < 1.1
+1.0 < 1.0.1
+1.1 > 1.0.1
+
+# SemVer compatibility
+1.5 > 1.5-pre1
+1.5 = 1.5+foobar
+
+# SemVer incompatibility
+1.5 < 1.5-2
+1.5-pre10 > 1.5-pre2
+
+# Empty strings
+ = 
+1 > 
+ < 1
+
+# Check boundary between textual and prerelease
+a-a < a
```

Comment:
I believe this describes the correct behavior. Why are all impls but Go failing this...?

Reply:
@Infinidoge seems to have figured it out; in Go, `-` is treated specially and splits out a component, but all the other implementations look for *textual components* that are *already split off* that start with `-`.

Checking the spec, that *is* correct behavior, but I don't know if that's the *intended* behavior...